llama.cpp

```diff
@@ -8055,8 +8061,9 @@ static void llama_set_inputs(llama_context & lctx, const llama_batch & batch) {
             }
         }
     } else {
-        // non-causal attention attends only the tokens within the batch (i.e. the KV cache is not used)
+        // with causal attention, the mask needs to match the kv cache size
         const int64_t n_tokens = batch.n_tokens;
+        const int64_t n_stride = hparams.causal_attn ? kv_self.n : n_tokens;
```

Comment:
I don't understand this change. This branch of the `if` is for non-causal masks, but the comment was changed to refer causal masks

Reply:
Ah yeah that comment isn't quite right. Should be more like "If the model uses the kv cache, then the mask has to match the kv size". Basically when we're trying to do embeddings on models that still use the kv cache, we need to mask the unused parts of the kv context or it will corrupt the results.

Reply:
Just pushed a fix for this comment.